examples/benchmarks/bubblebench/common/src/types.ts

```diff
@@ -33,12 +33,22 @@ export interface IClient {
 	bubbles: IBubble[];
 }
 
+/**
+ * @internal
+ */
+export interface ITreeClient {
+	clientId: string;
+	color: string;
+	// Make the bubbles read-only, as SharedTree ArrayNodes are not compatible with JavaScript arrays for writing purposes.
+	bubbles: readonly IBubble[];
+}
+
 /**
  * @internal
  */
 export interface IAppState {
-	readonly localClient: IClient;
-	readonly clients: IArrayish<IClient>;
+	readonly localClient: IClient | ITreeClient;
+	readonly clients: IArrayish<IClient | ITreeClient>;
```

Comment:
This approach seems odd to me. If users of IAppState need to be able to mutate the bubble arrays, then this won't work, since they can't tell if they have an ITreeClient. If they don't need to mutate the bubble array, then this is needlessly complex as you could just make IClient's bubble array readonly.